crates/ide/src/goto_type_definition.rs

```diff
@@ -56,27 +54,46 @@ pub(crate) fn goto_type_definition(
 
         Some((ty, node))
     })?;
-    let adt_def = ty.autoderef(db).filter_map(|ty| ty.as_adt()).last()?;
 
-    let nav = adt_def.try_to_nav(db)?;
-    Some(RangeInfo::new(node.text_range(), vec![nav]))
+    let mut res = FxHashSet::default();
+    let mut workload = vec![ty.strip_references()];
+    while let Some(ty) = workload.pop() {
```

Comment:
I think it may make sense to use the same logic as the hover actions instead?

Reply:
Right, doing what hover does makes a lot more sense here. This also drops trait objects I just realized.